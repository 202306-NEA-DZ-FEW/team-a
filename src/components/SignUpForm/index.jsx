import { useFormik } from "formik";
import * as Yup from "yup";

import { useAuth } from "@/context/AuthProvider";

import Input from "../Input";
import SelectInput from "../SelectInput";
import SignupMethods from "../SignupMethods";

function SignUpForm({ states, t }) {
  const { signUp } = useAuth();
  const passwordRules = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}$/;
  const formik = useFormik({
    initialValues: {
      name: "",
      email: "",
      password: "",
      confirmPassword: "",
      phone: "",
      location: "",
    },
    validationSchema: Yup.object({
      name: Yup.string()
        .max(20, t("signUp:nameLength"))
        .required(t("signUp:nameRequired")),
      email: Yup.string()
        .email(t("signUp:emailValid"))
        .required(t("signUp:emailRequired")),
      password: Yup.string()
        .min(8, t("signUp:passwordLength"))
        .matches(passwordRules, t("signUp:passwordStrength"))
        .required(t("signUp:passwordRequired")),
      confirmPassword: Yup.string()
        .oneOf([Yup.ref("password"), null], t("signUp:confirmPasswordMatch"))
        .required(t("signUp:confirmPasswordRequired")),
      phone: Yup.number().required(t("signUp:phoneRequired")),
      location: Yup.string().required(t("signUp:locationRequired")),
    }),
    onSubmit: async (values) => {
      const email = values.email;
      const password = values.password;
      const userInfo = { ...values };
      delete userInfo.password;
      delete userInfo.confirmPassword;
      await signUp(email, password, userInfo);
    },
  });

  return (
    <section className='flex gap-2 flex-col w-full mx-auto'>
      <h1 className='text-3xl font-bold'>{t("signUp:signUp")}</h1>
      <form onSubmit={formik.handleSubmit} className='flex flex-col gap-2'>
        <Input
          name='name'
          type='text'
          placeholder={t("signUp:namePlaceholder")}
          label={t("signUp:nameLabel")}
          handleChange={formik.handleChange}
          value={formik.values.name}
          handleBlur={formik.handleBlur}
          error={formik.errors.name}
          touched={formik.touched.name}
        />
        <Input
          name='email'
          type='email'
          placeholder={t("signUp:emailPlaceholder")}
          label={t("signUp:emailLabel")}
          handleChange={formik.handleChange}
          value={formik.values.email}
          handleBlur={formik.handleBlur}
          error={formik.errors.email}
          touched={formik.touched.email}
        />
        <Input
          name='password'
          type='password'
          placeholder={t("signUp:PasswordPlaceholder")}
          label={t("signUp:passwordLabel")}
          handleChange={formik.handleChange}
          value={formik.values.password}
          handleBlur={formik.handleBlur}
          error={formik.errors.password}
          touched={formik.touched.password}
        />
        <Input
          name='confirmPassword'
          type='password'
          placeholder={t("signUp:confirmPasswordPlaceholder")}
          label={t("signUp:confirmPasswordLabel")}
          handleChange={formik.handleChange}
          value={formik.values.confirmPassword}
          handleBlur={formik.handleBlur}
          error={formik.errors.confirmPassword}
          touched={formik.touched.confirmPassword}
          autoComplete='off'
        />
        <Input
          name='phone'
          type='text'
          placeholder={t("signUp:phonePlaceholder")}
          label={t("signUp:phoneLabel")}
          handleChange={formik.handleChange}
          value={formik.values.phone}
          handleBlur={formik.handleBlur}
          error={formik.errors.phone}
          touched={formik.touched.phone}
        />
        <SelectInput
          name='location'
          data={states}
          label={t("signUp:locationLabel")}
          handleChange={formik.handleChange}
          value={formik.values.location}
          touched={formik.touched.location}
          error={formik.errors.location}
        />
        <button
          type='submit'
          className='btn mt-2 btn-secondary text-black bg-opacity-40 w-full normal-case text-xl font-normal self-center rounded-xl'
        >
          {t("signUp:signUpButton")}
        </button>
      </form>
      <div className='lg:hidden'>
        <span className='divider'>{t("signUp:divider")}</span>
        <p className='text-center'>{t("signUp:signUpMethod")}</p>
        <SignupMethods />
      </div>
    </section>
  );
}

export default SignUpForm;
