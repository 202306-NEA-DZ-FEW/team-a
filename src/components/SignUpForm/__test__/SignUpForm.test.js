import renderer from "react-test-renderer";

import SignUpForm from "..";

it("renders correctly", () => {
  const t = jest.fn();
  const mockStates = [{ stateKey: "1", name: "adrar" }];
  const tree = renderer
    .create(<SignUpForm states={mockStates} t={t} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
