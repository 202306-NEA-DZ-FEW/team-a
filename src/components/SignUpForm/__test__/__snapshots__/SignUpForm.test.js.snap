// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="flex gap-4 flex-col w-full md:max-w-xl px-4"
>
  <h1
    className="text-3xl font-bold text-center"
  />
  <form
    className="flex flex-col gap-4"
    onSubmit={[Function]}
  >
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="name"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="email"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="password"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="confirmPassword"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="confirmPassword"
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div>
      <div
        className="label ml-2"
      >
        <label
          className="label-text"
          htmlFor="phone"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <input
        className="input bg-white input-primary w-full rounded-full border-opacity-25"
        id="phone"
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <div
        className="label"
      >
        <label
          className="label-text"
          htmlFor="location"
        />
        <label
          className="label-text-alt "
        />
      </div>
      <select
        className="select select-primary bg-white w-full border-opacity-25 rounded-3xl"
        defaultValue=""
        id="location"
        onChange={[Function]}
      >
        <option>
          adrar
        </option>
      </select>
    </div>
    <button
      className="btn btn-primary w-full rounded-3xl md:self-center text-white mt-8"
      type="submit"
    />
  </form>
  <span
    className="mt-8 divider"
  />
  <p
    className="text-center mb-4"
  />
  <div
    className="flex gap-4 justify-center items-center text-3xl text-primary"
  >
    <button
      className="btn btn-circle btn-sm bg-white"
    >
      <svg
        className="text-3xl text-blue-600"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 8.049c0-4.446-3.582-8.05-8-8.05C3.58 0-.002 3.603-.002 8.05c0 4.017 2.926 7.347 6.75 7.951v-5.625h-2.03V8.05H6.75V6.275c0-2.017 1.195-3.131 3.022-3.131.876 0 1.791.157 1.791.157v1.98h-1.009c-.993 0-1.303.621-1.303 1.258v1.51h2.218l-.354 2.326H9.25V16c3.824-.604 6.75-3.934 6.75-7.951z"
        />
      </svg>
    </button>
    <button
      className="btn btn-circle btn-sm bg-white"
    >
      <svg
        className="text-3xl text-primary"
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.1"
        viewBox="0 0 48 48"
        width="1em"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
          fill="#FFC107"
        />
        <path
          d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
          fill="#FF3D00"
        />
        <path
          d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
          fill="#4CAF50"
        />
        <path
          d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
          fill="#1976D2"
        />
      </svg>
    </button>
    <button
      className="btn btn-circle btn-sm bg-white"
    >
      <svg
        className="text-3xl text-gray-700"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
        />
      </svg>
    </button>
  </div>
</section>
`;
